Doc/c-api/init_config.rst

```diff
@@ -1605,6 +1605,71 @@ customized Python always running in isolated mode using
 :c:func:`Py_RunMain`.
 
 
+Get and Set the current runtime Python configuration
+====================================================
+
+The configuration option *name* parameter must be a non-NULL null-terminated
+UTF-8 encoded string.
+
+Some options are read from the :mod:`sys` attributes. For example, the option
+``"argv"`` is read from :data:`sys.argv`.
+
+
+.. c:function:: PyObject* PyConfig_Get(const char *name)
+
+   Get the current runtime value of a configuration option as a Python object.
+
+   * Return a new reference on success.
+   * Set an exception and return ``NULL`` on error.
+
+   The object type depends on the configuration option. It can be:
+
+   * ``bool``
+   * ``int``
+   * ``str``
+   * ``list[str]``
+   * ``dict[str, str]``
+
+   The caller must hold the GIL. The function cannot be called before
+   Python initialization nor after Python finalization.
+
+   .. versionadded:: 3.14
+
+
+.. c:function:: int PyConfig_GetInt(const char *name, int *value)
+
+   Similar to :c:func:`PyConfig_Get`, but get the value as a C int.
+
+   .. versionadded:: 3.14
+
+
+.. c:function:: PyObject* PyConfig_Names(void)
+
+   Get all configuration option names as a ``frozenset``.
+
+   Return a new reference on success.
+   Set an exception and return ``NULL`` on error.
+
+   The caller must hold the GIL.
+
+   .. versionadded:: 3.14
+
+
+.. c:function:: int PyConfig_Set(const char *name, PyObject *value)
+
+   Set the current runtime value of a configuration option.
+
+   * Raise a :exc:`ValueError` if there is no option *name*.
+   * Raise a :exc:`ValueError` if *value* is an invalid value.
```

Comment:
For example, pass a negative number when a bool is expected.

Reply:
IIRC you accept >= 0 for bools but not < 0 right? maybe this could be added in the docs (maybe add something after the `bool` bullet point saying that anything >= 0 is considered as boolean but not <= 0?):

```rst
   The object type depends on the configuration option. It can be:
   
   * ``bool`` (nonnegative integer)
```

Something like that?